### ISSUE TITLE
False positive for rule CA1002 on sealed classes

### ISSUE THREAD

**Issue by EvinWatson** [dotnet/runtime#88310]
### Analyzer

**Diagnostic ID**: [CA1002](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1002): `Do not expose generic lists`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 7.0.0

### Describe the bug

This rule prevents the use of Generic Lists for inheritance issues, however sealed classes cannot be inherited, so the rule should not apply. I suggest creating an exception for sealed classes.

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/sealed

### Steps To Reproduce

1. Create a 'sealed' class.
2. Create a property of type Generic List.
3. Observe the analyzer give error CA1002.

### Expected behavior
Analyzer should not cite CA1002 on sealed classes.

### Actual behavior
Analyzer cites CA1002 on sealed classes.

## Additional context
None

**Comment by ericstj** [comment]
Moving to `System.Collections` area since this is feedback on an analyzer that applies to use of collections.  cc @buyaa-n

**Comment by eiriktsarpalis** [comment]
@buyaa-n is this something we should consider addressing for 8.0?

**Comment by buyaa-n** [comment]
> @buyaa-n is this something we should consider addressing for 8.0?

There is an [open PR](https://github.com/dotnet/roslyn-analyzers/pull/6729/files) already in the roslyn-analyzers repo, so we can address it in 8.0. It seems the issue moved here for deciding if we support the suggested fix. For me the suggestion `CA1002 Analyzer should not flag on sealed classes` makes sense. 

Let me know what you think @eiriktsarpalis @stephentoub

**Comment by Frulfump** [comment]
> I believe the rule description is talking about inheritance for the collection itself, and not the element type of the collection: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1002#rule-description
> 
> I'll move this issue to dotnet/runtime to get it triaged. 

I was under this impression as well?!

**Comment by stephentoub** [comment]
> I believe the rule description is talking about inheritance for the collection itself, and not the element type of the collection

Yeah, I'm not understanding what is supposed to change here. As shown in the examples in https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1002, the issue is about the type of the property itself, that `List<T>` (regardless of what `T` is) is undesirable for the result of a property because the containing type (regardless of whether it's sealed or not) can't change the behavior of that collection, e.g. it can't modify the behavior of the `List<T>` to throw if someone tries to store `null` or can't track mutations and so on.

I'm going to close this. If I've misunderstood the concern, please feel free to re-open.
